core/src/main/java/org/elasticsearch/action/admin/indices/shards/IndicesShardsStoresRequest.java

```diff
@@ -0,0 +1,193 @@
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.action.support.master.MasterNodeReadRequest;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class IndicesShardsStoresRequest extends MasterNodeReadRequest<IndicesShardsStoresRequest> implements IndicesRequest.Replaceable {
+
+    /**
+     * Status used to choose shards to get store information on
+     */
+    public enum Status {
+
+        /**
+         * Status to get store information on all active shards
+         */
+        GREEN((byte) 0),
+
+        /**
+         * Status to get store information on shards with
+         * at least one unassigned replica
+         */
+        YELLOW((byte) 1),
+
+        /**
+         * Status to get store information on shards with
+         * unassigned primary or replica
+         */
+        RED((byte) 2),
+
+        /**
+         * Status to get store information on all shards
+         */
+        ALL((byte) 3);
+
+        private final byte id;
+
+        Status(byte id) {
+            this.id = id;
+        }
+
+        public byte id() {
+            return id;
+        }
+
+        public static Status fromId(byte id) {
+            switch (id) {
+                case 0:
+                    return GREEN;
+                case 1:
+                    return YELLOW;
+                case 2:
+                    return RED;
+                case 3:
+                    return ALL;
+                default:
+                    throw new IllegalArgumentException("No status for id [" + id + "]");
+            }
+        }
+
+        public static Status fromString(String status) {
+            if (status.equalsIgnoreCase("green")) {
+                return GREEN;
+            } else if (status.equalsIgnoreCase("yellow")) {
+                return YELLOW;
+            } else if (status.equalsIgnoreCase("red")) {
+                return RED;
+            } else if (status.equalsIgnoreCase("all")) {
+                return ALL;
+            } else {
+                throw new IllegalArgumentException("unknown status [" + status + "]");
+            }
+        }
+
+        public static Status readFrom(StreamInput in) throws IOException {
+            return fromId(in.readByte());
+        }
+
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeByte(id());
+        }
+    }
+
+    private String[] indices = Strings.EMPTY_ARRAY;
+    private IndicesOptions indicesOptions = IndicesOptions.strictExpandOpenAndForbidClosed();
+    private Status[] shardStatuses = new Status[] { Status.YELLOW };
```

Comment:
can we use an EnumSet<ClusterHealthStatus> ? We can then let the Status enum go ...right?